### PR TITLE
Module6

### PR DESCRIPTION
**Pull request by avdhut1977**
Module6 Project diagrams

### CODE REVIEW

**Comment by BenRamchandani** [comment]
Hi Avdhut,
Can I get you to add some more detail to the diagrams? Someone looking at the diagrams without any knowledge of it should be able to get an idea of how the app fits together and interacts with other systems.

Add details to the boxes and arrows to make it clear what each component does and how they interact. Some specific pointers:
* Context diagram
  * Label each arrow, for example the User "Creates and manages to do items"
  * Make it clear that Trello is external and not part of our app
* Container diagram
  * Add detail to each box, e.g. what does Flask do vs. our code?
  * How does it interact with Trello?
* Component diagram
  * Add more detail. What does each part do?

**Comment by avdhut1977** [comment]
Implemented the review comments for Module6 and Module5 Stretched goal for Module7

File: Dockerfile [hunk 1]
```diff
@@ -14,3 +14,15 @@ ENV FLASK_ENV=production
 COPY . /app/
 ENTRYPOINT poetry run gunicorn --bind 0.0.0.0:5000 todo_app.wsgi:app
 
+FROM base as test
+RUN apt-get update && apt-get install curl -y
+RUN apt-get update && apt-get install wget -y
+#Install chrome
+RUN ls -lrt /etc/apt
+RUN apt-get update && apt-get install -y gnupg2
```
Comment (BenRamchandani) [inline Dockerfile:22]: Can simplify this, which will make the image build faster
```suggestion
RUN apt-get update && apt-get install curl wget gnupg2 -y
```

File: todo_app/test_system.py [hunk 2]
```diff
@@ -52,17 +55,24 @@ def test_app():
...
+#def driver():     
+#    with webdriver.Chrome() as driver:
+#        yield driver 
+def driver():
+    opts = Options()
+    opts.add_argument('--headless')
+    opts.add_argument('--no-sandbox')
+    opts.add_argument('--disable-dev-shm-usage')
+    with webdriver.Chrome(ChromeDriverManager().install(), options=opts) as driver: #newly added
+    #with Chrome() as driver:
```
Comment (BenRamchandani) [inline todo_app/test_system.py:67]: Can delete the commented out code

**Review by BenRamchandani — APPROVED** [review]
The diagrams look good, thanks for adding more detail :+1:

The testing in Docker changes look sensible too. For module 7 they'll need to run on Travis which means the .env file will not be available. Instead of using a bind mount to run your tests you're better off using the `--env-file` option locally, then on Travis you can add everything as environment variables and do `docker run -e TRELLO_API_KEY -e TRELLO_API_SECRET ...`.
That means you'll want to run your e2e tests separately in docker, it can be the same image, but a different run command as they need different environment variables.

One other thing: can you add placeholders for the various Trello environment variables to .env.template? It makes our job easier when we're trying to get it running locally!

**Pull request merged**